bin/convencommitcheck.bash

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+
+# Define valid types and scopes
+VALID_TYPES=("build" "chore" "ci" "docs" "feat" "fix" "perf" "refactor" "style" "test")
+VALID_SCOPES=("dev" "comp" "lib" "fd2" "examples" "school" "none") # Added 'none' for no scope
+
+# Helper function to check if an element is in an array
+elementInArray() {
+    local element
+    for element in "${@:2}"; do
+        [[ "$element" == "$1" ]] && return 0
```

Comment:
Technically for conventional commits these comparisons should be case insensitive.

Here's more information: https://www.conventionalcommits.org/en/v1.0.0/#specification

Reply:
Intergrated shopt for case insensitive matching during script runtime (turns it off after)